src/library/autodj/autodjprocessor.cpp

```diff
@@ -1078,7 +1090,17 @@ double AutoDJProcessor::getLastSoundPosition(DeckAttributes* pDeck) {
             return samplePositionToSeconds(lastSound, pDeck);
         }
     }
-    return pDeck->trackTime();
+    return getEndPosition(pDeck);
+}
+
+double AutoDJProcessor::getEndPosition(DeckAttributes* pDeck) {
```

Comment:
The "play position" control is normalized to [0.0, 1.0], but "end position" is measured in seconds??

We should not use different scales when referring to some general "position"!

Reply:
Yes right. I would love to fix this. But this is too risky for now and in any case out of scope of this PR.

Reply:
What's the risk to correctly name this newly introduced function `getEndPositionSeconds()`? Otherwise remove the final invocation of `samplePositionToSeconds(), then the naming would be valid.

Reply:
Ah, that's what you mean. I will fix this and rename the sibling functions as well.

Reply:
Done. Unfortunately the underlying issue remains. But that is a topic of another PR.